client/src/entrypoints/admin/core.js

```diff
@@ -318,15 +318,20 @@ $(() => {
         $inputContainer.addClass(workingClasses);
         searchNextIndex++;
         const index = searchNextIndex;
+
+        // Only update q and keep other query params
```

Comment:
Should we switch back to the first page when the search query changes?

Reply:
Good call. Fixed.